src/Loading/sceneLoader.ts

```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
+     */
+    public static ImportAnimationsAsync(rootUrl: string, sceneFilename: string | File = "", scene: Nullable<Scene> = EngineStore.LastCreatedScene, targetConverter: Nullable<(target: any) => any> = null, onSuccess: Nullable<(scene: Scene) => void> = null, onProgress: Nullable<(event: SceneLoaderProgressEvent) => void> = null, onError: Nullable<(scene: Scene, message: string, exception?: any) => void> = null): Promise<Scene> {
+        return new Promise((resolve, reject) => {
+            SceneLoader.ImportAnimations(rootUrl, sceneFilename, scene, targetConverter, (_scene: Scene) => {
+                resolve(_scene);
+            }, onProgress, (_scene: Scene, message: string, exception: any) => {
+                reject(exception || new Error(message));
+            });
+        });
+    }
+
+    /**
+     * Merge animations from an asset container into a scene
+     * @param scene is the instance of BABYLON.Scene to append to (default: last created scene)
+     * @param animationAssetContainer is the instance of BABYLON.AssetContainer containing animations
+     * @param targetConverter defines a function used to convert animation targets from the asset container to the scene (default: search node by name)
+     */
+    public static MergeAnimations(scene: Scene, animationAssetContainer: AssetContainer, targetConverter: Nullable<(target: any) => any> = null): void {
+
+        let _targetConverter = targetConverter ? targetConverter : this._defaultTargetConverter(scene);
+
+        // Copy node and bone animations
+        let animatableObjectsInAC = this._getAllAnimatableObjects(animationAssetContainer);
+        animatableObjectsInAC.forEach(animatableObjectInAC => {
+            let objectInScene = _targetConverter(animatableObjectInAC);
+            if (objectInScene != null) {
+                objectInScene.animations = animatableObjectInAC.animations;
+            }
+        });
+
+        // Copy animation groups
+        animationAssetContainer.animationGroups.slice().forEach(animationGroupInAC => {
+            // Dispose animation groups with same name as one being loaded
+            scene.animationGroups.forEach(animationGroup => {
+                if (animationGroup.name == animationGroupInAC.name) {
+                    animationGroup.dispose();
+                }
+            })
+
+            // Clone the animation group and all its animatables
+            animationGroupInAC.clone(animationGroupInAC.name, _targetConverter);
+
+            // Remove animatables related to the animation asset container
+            animationGroupInAC.animatables.forEach(animatable => {
+                animatable.stop();
+            })
+        });
+
+        // Copy animatables
+        scene.animatables.slice().forEach(animatable => {
+            let target = _targetConverter(animatable.target);
+
+            // If the animatable has just been loaded
+            if (target && target != animatable.target) {

```

Comment:
```suggestion
            if (target && target !== animatable.target) {
```

Reply:
Ok

Reply:
Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/7c892e0aae5ae7606fa23989738d49c61e182b6b